File: Data Structures & Algorithms/validate-parentheses/submission-2.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        for c in s:
            if c == '(' or c == '{' or c == '[':
                stack.append(c)
            elif len(stack) > 0:
                open_bracket = stack.pop()
                if (c == ')' and open_bracket != '(') or (c == '}' and open_bracket != '{') or (c == ']' and open_bracket != '['):
                    return False
            else:
                return False
        return True

File: Data Structures & Algorithms/minimum-stack/submission-2.java
class MinStack {

    private Stack<Integer> stackOne;
    private Stack<Integer> stackTwo;

    public MinStack() {
        stackOne = new Stack<Integer>();
        stackTwo = new Stack<Integer>();
    }
    
    public void push(int val) {
        stackOne.push(val);
        if (stackTwo.empty() || val <= stackTwo.peek()) {
            stackTwo.push(val);
        }
    }
    
    public void pop() {
        int val = stackOne.pop();
        if (stackTwo.peek() == val) {
            stackTwo.pop();
        }
    }
    
    public int top() {
        return stackOne.peek();
    }
    
    public int getMin() {
        return stackTwo.peek();
    }
}
